app/models/observation.rb

```diff
@@ -193,6 +193,61 @@ class Observation < AbstractModel
   # Automatically (but silently) log destruction.
   self.autolog_events = [:destroyed]
 
+  include ScopesForTimestamps
+
+  # NOTE: Experimental; not tested yet. Needs tests.
+  scope :of_name, lambda {
+    |name,
+     include_synonyms = false,
+     include_subtaxa = false,
+     include_all_name_proposals = false,
+     of_look_alikes = false|
+
+    names = [name]
+    names = name.synonyms if include_synonyms
+    if include_subtaxa
+      names.each do |n|
+        names << Name.subtaxa_of(n)
+      end
+    end
+
+    if include_all_name_proposals
+      joins(:namings).where(namings: { name: names })
+    elsif of_look_alikes
+      joins(:namings).where(namings: { name: names }).
+        where(Observation[:name] != name)
+    else
+      where(name: names)
+    end
+  }
+
+  scope :of_name_like,
+        ->(name) { where(name: Name.text_name_includes(name)) }
+  scope :with_name, -> { where.not(name: Name.unknown) }
+  scope :without_name, -> { where(name: Name.unknown) }
+  scope :by_user, ->(user) { where(user: user) }
+  scope :at_location, ->(location) { where(location: location) }
+  scope :in_region,
+        ->(where) { where(Observation[:where].matches("%#{where}%")) }
+  scope :is_collection_location, -> { where(is_collection_location: true) }
+  scope :not_collection_location, -> { where(is_collection_location: false) }
+  scope :for_project, lambda { |project|
+    joins(:project_observations).
+      where(ProjectObservation[:project_id] = project)
+  }
+  scope :with_image, -> { where.not(thumb_image: nil) }
+  scope :without_image, -> { where(thumb_image: nil) }
+  scope :with_location, -> { where.not(location: nil) }
+  scope :without_location, -> { where(location: nil) }
+  scope :notes_include,
+        ->(notes) { where(Observation[:notes].matches("%#{notes}%")) }
+  scope :with_notes, -> { where.not(notes: Observation.no_notes) }
+  scope :without_notes, -> { where(notes: Observation.no_notes) }
+  scope :with_sequence, -> { where(sequence: true) }
+  scope :without_sequence, -> { where(sequence: false) }
```

Comment:
How about AR
```ruby
missing(:sequences)
```
I just stumbled across `missing`, which is new in AR 6.1. https://api.rubyonrails.org/v6.1/classes/ActiveRecord/QueryMethods/WhereChain.html#method-i-missing

Reply:
I like this!

Reply:
And it gives us something we can't get through the UI. The tradeoff is an outer join.

Reply:
That's a tradeoff for sure. Hm.

Reply:
Revised! That's the only way to do it! 

Pending your response on whether the scope is worth it...

Reply:
I think they are both worth it.
The pattern search UI does include `sequence: yes`
which does
```sql
QUERY starting: "\n      SELECT DISTINCT observations.id\n      FROM `observations` JOIN `sequences` ON sequences.observation_id = observations.id\n      ORDER BY observations.when DESC, observations.id DESC\n"
   (16.4ms)  
      SELECT DISTINCT observations.id
      FROM `observations` JOIN `sequences` ON sequences.observation_id = observations.id
      ORDER BY observations.when DESC, observations.id DESC
  LIMIT 0,90
```
So we're already doing the inner join.

But we don't allow `sequence: no`, perhaps because of the cost.
I personally think it's also worth having. @pellaea might disagree.

Reply:
Committed!

Reply:
Yes,to do the "no sequence" equivalent requires a left outer join, and I am told that is much more expensive, although I admit that I never actually tested it.

Reply:
Ok. Well, we have the scope as of now, but we certainly don't have to call it in the UI, and i can't quite imagine why we'd need to. It might come in handy in writing tests or queries — or not. 

I kind of like having it on hand just to remind me that `missing: :sequence` is new valid AR. I wonder if AR will soon have a comparator for `matches`!